trace-server/org.eclipse.tracecompass.incubator.trace.server.jersey.rest.core/src/org/eclipse/tracecompass/incubator/internal/trace/server/jersey/rest/core/model/BookmarkQueryParameters.java

```diff
@@ -0,0 +1,40 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Ericsson
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License 2.0 which
+ * accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.tracecompass.incubator.internal.trace.server.jersey.rest.core.model;
+
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.media.Schema.RequiredMode;
+
+/**
+ * Parameters for bookmark creation and update operations
+ *
+ * @author Kaveh Shahedi
+ * @since 10.1
+ */
+public interface BookmarkQueryParameters {
```

Comment:
This doesn't reflect the actual implementation. The implemenation uses QueryParameters which returns the parameter map. Your implementation uses the parameter map. Alternative you could have a implementation that pre-parses the JSON received over the network and hence you can have get the name, start and end directly without parsing the parameters map. In most cases in the server, the parameter map is used. The ConfigurationManagerService is done a bit differently, where some default TSP parameters (e.g. name or description) are defined and parsed. 

Besides, I don't see the `value` parameter anywhere.

Reply:
I investigated this comment, but I couldn't figure out what should be done. I followed the same structure that is present in `ExperimentManagerService` class. Specifically:
```java
public Response postExperiment(@RequestBody(content = {
            @Content(schema = @Schema(implementation = ExperimentQueryParameters.class))
    }, required = true) QueryParameters queryParameters)
```
As it is implemented here, the `postExperiment` is using `ExperimentQueryParameters` schema, but also receiving a `QueryParameters` as argument. 

Also, I didn't understand the `value` parameter issue. Can you explain it a bit more?

P.S.: I ran the swagger yaml generation process, and it was successful.

Reply:
It looks like `ExperimentQueryParameters` swagger schema is incorrect and doesn't match the actual implementation. I should have bee defined similarly to `RequestedQueryParameters`.

The implementation for experiments ask for the following input:
```
{
    parameters: {
       "name": "experiment name",
       "traces: ["uuid1", "uuid2"]
    }
}
```

but the actual swagger documentation says:

```
{
     "name": "experiment name",
     "traces: ["uuid1", "uuid2"]
}
```

So, if you'd like to use `parameters` key as you have implemented in the `BookmarkServiceManager`, then swagger documentation needs to be changed. Alternatively, you adjust the implementation to match the documentation. Maybe the first option is better, to be consistent with other endpoints.

About the comment about `value`, I'm not sure anymore, maybe it was about the `payload` that is part of the query parameters but there is no documentation.